### ISSUE TITLE
Add tblproperties using alter table with spark

### ISSUE THREAD

**Issue by AlvaroNavarroMora** [apache/iceberg#6105]
### Apache Iceberg version

0.13.1

### Query engine

Spark

### Please describe the bug 🐞

We are trying to add the following properties to existing iceberg tables using spark 3.1.2:

```
'write.metadata.delete-after-commit.enabled'='true'
'write.metadata.previous-versions-max'='10'
```

In order to do so we are using the next queries:
```
ALTER TABLE <tableName> SET TBLPROPERTIES ('write.metadata.delete-after-commit.enabled'='true')
ALTER TABLE <tableName> SET TBLPROPERTIES ('write.metadata.previous-versions-max'='10')
```

The queries are being executed without any errors, but after reviewing the tables the properties we observe that the properties have not changed.

We can't add the properties when creating the table because we are using liquibase with the athena connector, which does not support Iceberg specific properties. This is why we need to add the properties afterwards.

Is it possible to add these properties to existing tables using spark in any other way? if that is the case, how should we do it?

**Comment by nastra** [comment]
@AlvaroNavarroMora I don't see any reason why those table properties wouldn't be reflected/set. Are you running `SHOW TBLPROPERTIES <tblName>` or something else to describe the table properties?

**Comment by ELHoussineT** [comment]
@AlvaroNavarroMora Did it work for you?

**Comment by AlvaroNavarroMora** [comment]
Hi, after doing some testing we have observed that the properties are being applied to the tables, however it seems to not be working properly. We have tables with those properties applied that have over 4k metadata files.

**Comment by nastra** [comment]
The same issue came up in a [Slack discussion](https://apache-iceberg.slack.com/archives/C025PH0G1D4/p1668714478198079?thread_ts=1668645354.671479&cid=C025PH0G1D4) and I've opened https://github.com/apache/iceberg/pull/6217 to address that.

**Comment by ELHoussineT** [comment]
@AlvaroNavarroMora Interesting, we did as the example you shared in the main thread and we saw that the metadata were deleted. For verification of the deletion of the metadata, we used pyiceberg from this PR https://github.com/apache/iceberg/pull/6034. 

Running the following code before and after setting the table properties showed that the old metadata were in fact deleted. 

```
from pyiceberg.catalog.glue import GlueCatalog
catalog = GlueCatalog("glue", warehouse=f"s3://...")
table = catalog.load_table("<glue db>.<table>")
print(table.metadata.metadata_log)
```
